scipy/stats/_stats_py.py

```diff
@@ -7231,13 +7239,23 @@ def chisquare(f_obs, f_exp=None, ddof=0, axis=0):
     the asymptotic distribution is not chi-square, in which case this test
     is not appropriate.
```

Comment:
Suggestion: However, it is also possible that the asymptotic distribution for an arbitrary estimation is not chi-square (for example, because the estimation is not consistent), in which case this test is not appropriate.

Reply:
I think it should be made clear that the statistic is asymptotically always chi-square distributed if one indeed passes observed and expected frequencies, and no estimation is happening. Deviations from an asymptotic chi-square distribution can occur if the estimation is not consistent.

Reply:
Thanks. Since that's part of the existing docstring and not altered here, I'll leave that for another pull request.

Reply:
I'd advocate for not getting into the weeds here - we're not trying to provide a textbook on statistical theory?  The caveat that "this test may not be appropriate" applies to any procedure at all. If you've got a particular gotcha that's easy to explain in mind, though, that could be useful? (Note that in my edit above I mentioned that it's testing whether the observations are independent, which is the main source of discrepancy; we could explain more but maybe anyone who's sophisticated enough to pick up on that will already be familiar with the chi-squared test?)

Reply:
I understand that the text not become too long, but in an ideal world, we should fully specify when exactly the test is applicable. We cannot make strong statements about finite samples, but we can fully specify when the test is applicable in the asymptotic regime (namely when the conditions are met under which it was originally derived). I think that would take little space.

Firstly, it is common in practice people apply statistical tests and methods to situations where they are not valid, because - understandably - one has to study the theory and read a lot of text books to really understand everything.

Secondly, we should be precise. This is a scientific library about applied math. It should be all about correctness. People often perceive statistics as a messy branch of math, but it is not messy at all. It is as well founded in proofs as any other branch of math.